リリース前レビュー指摘対応：報告一覧ページのCSVエクスポート機能追加

shoutarou123 opened the pull request:
### 概要
　リリース前レビュー指摘対応：報告一覧ページのCSVエクスポート機能追加修正完了

### タスク
- [ ] なし
- [x] あり https://docs.google.com/spreadsheets/d/13gLhBRmXqtZu4EBsfG-Jq9vD32EhqQzGTKOQlgkpb4U/edit?usp=sharing

### 実装内容・手法
 各報告一覧からCSVエクスポート出力するよう実装

### gemfileの変更
- [x] なし
- [ ] あり

shoutarou123 committed:
reports_controller 各報告CSV取得のコード追加

shoutarou123 committed:
各報告のCSVエクスポートボタン追加

shoutarou123 committed:
reports_controller 検索データをセッションに保存するよう追加

shoutarou123 committed:
ｲﾝﾃﾞﾝﾄ調整

shoutarou123 committed:
reports_controller 構文チェックによるリファクタリング

shoutarou123 committed:
reports_controller reports_by_searchﾒｿｯﾄﾞ 検索条件ｾｯｼｮﾝに保存するｺｰﾄﾞ追加。よってｾｯｼｮﾝｸﾘｱ時のprevious_searchが機能する。

shoutarou123 committed:
reports_controller reports_by_searchﾒｿｯﾄﾞ 検索ﾊﾟﾗﾒｰﾀｰの存在性記述を簡素化

shoutarou123 committed:
CSVボタンマージン微調整

kousuke-yamada commented:
以下の動作を行った時、表示されている報告内容と、エクスポートされる報告内容が異なっているようです。
① 報告一覧ページにて、キーワード検索を行う
② 報告一覧ページから別ページに遷移
③ 再度、報告一覧ページに遷移
④ 報告一覧ページににて、エクスポートを行う。

○通常のエクスポート
<img width="1427" alt="スクリーンショット 2024-08-19 8 31 44" src="https://github.com/user-attachments/assets/d85b2a95-d230-47b3-9829-b95cc8a9e626">
<img width="592" alt="スクリーンショット 2024-08-19 8 32 13" src="https://github.com/user-attachments/assets/47c4b8aa-ebf8-4468-9e87-62df8fbab4f4">

○検索後のエクスポート
<img width="1432" alt="スクリーンショット 2024-08-19 8 32 28" src="https://github.com/user-attachments/assets/f2d2029d-54d4-4b56-976f-57c1ad7f6604">
<img width="562" alt="スクリーンショット 2024-08-19 8 32 35" src="https://github.com/user-attachments/assets/b5f967c8-f57a-4f99-bbd5-cc82c751cf80">

○検索後、別ページに遷移して、再度報告一覧ページに遷移して通常エクスポート
<img width="1430" alt="スクリーンショット 2024-08-19 8 32 46" src="https://github.com/user-attachments/assets/b6e17348-acbb-4bc2-b041-b3e09e4a1fc5">
<img width="502" alt="スクリーンショット 2024-08-19 8 32 53" src="https://github.com/user-attachments/assets/4c927baf-b651-4772-9a81-0ef76781e923">


セッションのクリアタイミングを再検討頂きたく。
ご確認よろしくお願い致します。
（※ マージしてしまいましたが、連絡の方も同様かもしれません。合わせてご確認お願いします。）

shoutarou123 committed:
reports_controller ｾｯｼｮﾝｸﾘｱ条件分け

shoutarou123 committed:
routes get :export_csv追加
2ﾍﾟｰｼﾞCSV取得するも1ﾍﾟｰｼﾞﾃﾞｰﾀ取得されるため追加したもの

shoutarou123 committed:
reports_controller routes追加により画面変更時のｾｯｼｮﾝｸﾘｱ削除
CSV専用ｱｸｼｮﾝ追加
不要なﾒｿｯﾄﾞ削除
indexｱｸｼｮﾝﾘﾌｧｸﾀﾘﾝﾝｸﾞ

shoutarou123 committed:
各CSVﾎﾞﾀﾝのﾊﾟｽ変更

shoutarou123 committed:
routes.rb export_csvの記述修正

shoutarou123 committed:
reports_controller indexにｾｯｼｮﾝｸﾘｱする記述追加

shoutarou123 committed:
reports_controller
記述位置修正、report_typeに応じてﾍﾟｰｼﾞﾈｰｼｮﾝ無視ID取得しｾｯｼｮﾝに保存する記述追加、そのﾃﾞｰﾀをCSVで取得する記述追加。

shoutarou123 committed:
今月の報告、今週の報告についてdata使用。この記述が無いとreport_type取得できずﾎﾞﾀﾝ機能せず、csv_typeもnilになってしまう。

shoutarou123 committed:
report_controller
検索一覧 ﾍﾟｰｼﾞﾈｰｼｮﾝ無視ﾃﾞｰﾀ取得しｾｯｼｮﾝに保存するｺｰﾄﾞ追加。

shoutarou123 committed:
reports_controller ﾘﾌｧｸﾀﾘﾝﾝｸﾞ

shoutarou123 committed:
reports_controller
reports_by_searchﾒｿｯﾄﾞ内のsession_save引数、fetch_full_reportsﾒｿｯﾄﾞをｲﾝｽﾀﾝｽ変数に修正(検索ﾎﾞﾀﾝが機能しないことによる修正)
